### ISSUE TITLE
Dependency Injection?

### ISSUE THREAD

**Issue by markphelps** [codeship/go-best-practice#28]
Possible section on how we do dependency injection (using interfaces, no framework, etc)?

**Comment by GeorgeMac** [comment]
Yeah I like it. Something short and sweet. Perhaps with some practical examples of small interfaces using composition and decoration.
